core/modules/rproc/InfileMerger.h

```diff
@@ -156,10 +157,18 @@ class InfileMerger {
     std::mutex _createTableMutex; ///< protection from creating tables
     std::mutex _sqlMutex; ///< Protection for SQL connection
     bool _needCreateTable{true}; ///< Does the target table need creating?
+    size_t _getResultTableSizeMB(); ///< Return the size of the result table in MB.
 
     mysql::MySqlConnection _mysqlConn;
+
     std::mutex _mysqlMutex;
     lsst::qserv::mysql::LocalInfile::Mgr _infileMgr;
+
+    std::string _queryIdStr{"QI=?"}; ///< Unknown until results start coming back from workers.
+
+    int _sizeCheckRowCount{0}; ///< Number of rows read since last size check.
+    int _checkSizeEveryXRows{1000}; ///< Check the size of the result table after every x number of rows.
+    size_t _maxResultTableSizeMB{5000}; ///< Max result table size.
```

Comment:
Why? I know there can issues with curly braces in constructor initialization lists, but being able to set the initial values in one place (instead of having to set it in every constructor) is a definite win.